src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,197 @@
+        }
+        final Map<String, String[]> checksNamesMap = getXdocsMap();
+        for (List<String> sub : list) {
+            final String moduleName = sub.get(0);
+            final String checkName1 = sub.get(1);
+            final String checkName2 = sub.get(2);
+            final String errorMessage =
+                    "URL in" + AVAILABLE_CHECKS_PATH + " is inconsistent: "
+                            + moduleName + " " + checkName1 + " " + checkName2;
+            assertEquals(checkName1, checkName2, errorMessage);
+            if (checkName1.equals(COMMENTS_INDENTATION) || checkName1.equals(INDENTATION)) {
+                assertEquals(MISC, moduleName, errorMessage);
+            }
+            else if (checkName1.equals(SUPPRESS_WARNINGS_HOLDER)) {
+                assertEquals(ANNOTATION, moduleName, errorMessage);
+            }
+            else {
+                final String[] moduleFileNames = checksNamesMap.get(moduleName);
+                if (moduleFileNames != null) {
+                    boolean match = false;
+                    for (String filename : moduleFileNames) {
+                        if ((checkName1 + SUFFIX_CHECK_JAVA).equals(filename)) {
+                            match = true;
+                            break;
+                        }
+                    }
+                    assertTrue(match, errorMessage);
+                }
+            }
+        }
+    }
+
+    public static final class CheckTest extends DefaultHandler {
+
+        public static final String SPLIT_ONE = "#";
+
+        public static final String SPLIT_TWO = "\\.";
+
+        public static final String PREFIX_CONFIG = "config_";
+
+        private final String nodeName;
+
+        private List<List<String>> list;
+
+        private List<String> stringList;
+
+        private String currentTag;
+
+        public CheckTest(String nodeName) {
+            this.nodeName = nodeName;
+        }
+
+        public List<List<String>> getList() {
+            final List<List<String>> copyList = new ArrayList<>();
+            for (List<String> elemList: list) {
+                final List<String> tempList = new ArrayList<>(elemList);
+                copyList.add(tempList);
+            }
+            return copyList;
+        }
+
+        @Override
+        public void startDocument() {
+            list = new ArrayList<>();
+        }
+
+        @Override
+        public void startElement(String uri, String localName, String qName,
+                                 Attributes attributes) {
+            if (qName.equals(nodeName)) {
+                final String[] splitStrings = attributes.getValue(0).split(SPLIT_ONE);
+                if (splitStrings[0].substring(0, PREFIX_CONFIG.length()).equals(PREFIX_CONFIG)) {
+                    stringList = new ArrayList<>();
+                    final String moduleName =
+                            splitStrings[0].substring(PREFIX_CONFIG.length()).split(SPLIT_TWO)[0];
```

Comment:
`splitStrings[0].substring(PREFIX_CONFIG.length()).split(SPLIT_TWO)[0];`
It is not clear how module name is extracted.

Reply:
done

Reply:
It is not clear.
The reason I am asking is that variable names like `splitStrings`, `SPLIT_ONE` and `SPLIT_TWO` tells me nothing about logic, it is hard to follow.

Reply:
done